ci-metrics/README.md

```diff
@@ -0,0 +1,171 @@
+
+```
+		   +----------+
+		   |   ELK    |                                   +--------------------+
+		   | instance |                                   |     Jenkins 1      |
+		   +----+-----+                   +---------------+   with test jobs   |
+				^                         |               | using CI publisher |
+				|                         |               +--------------------+
+				|                         |
+		+-------+---------+               v               +--------------------+
+		|  Jenkins with   |      +--------+-------+       |     Jenkins 2      |
+		| CI Metrics data +<-----+ CI Message Bus +<------+   with test jobs   |
+		|  listener job   |      +--------+-------+       | using CI publisher |
+		+-----------------+               ^               +--------------------+
+										  |
+										  |               +--------------------+
+										  |               |     Jenkins 3      |
+										  +---------------+   with test jobs   |
+														  | using CI publisher |
+														  +--------------------+
+```
+
+The ELK instance is the data store for storing metrics exported from Jenkins jobs with enabled CI publisher. The exported data from Jenkins jobs is harvested via CI Metrics data listener Jenkins job and stored into the ELK instance.
+
+If you want to install your own ELK instance follow the steps in the section ELK instance.
+
+If you do not want to install you own ELK instance skip the sections 'ELK instance' and 'CI Metrics data listener Jenkins job' sections.
+
+
+## ELK instance
+
+### What is ELK
+
+ELK is a combination of Elastic Search, Logstash and Kibana. Elasticsearch is a non relational database. Logstash collects and pushes logs to the Elasticsearch database. Kibana then takes the data and visualizes it in graphical form and it also makes it possible to query the data.
+
+Note that in current design Logstash is not used and the data is directly stored from the CI Metrics data listener Jenkins job into Elasticsearch.
+
+### How to install ELK for CI metrics
+
+Note that the solution requires an RHEL 7 or Centos 7 system. To install an ELK instance on your localhost use the provided Ansible yaml file.
+
+
+1. ####Add EPEL repository
+
+  See [https://fedoraproject.org/wiki/EPEL] for installation intructions
+
+
+1. ####Install git and ansible
+
+  >$ yum -y install git ansible
+
+
+1. ####Get CI Metrics files
+
+  >$ git clone https://github.com/RHQE/platform-ci.git
+
+  >$ cd platform-ci/ci-metrics
+
+
+1. ####Run ansible to install ELK
+
+  >$ ansible-playbook install_elk.yaml
+
+
+1. ####Verify that ELK instance works
+
+    Access the port 80 on your localhost via your web browser and you should see the Kibana configuration page.
+
+  [http://localhost]
+
+
+## ELK Listener Jenkins job
+
+ELK Listener job listens on the CI message bus for messages matching this JMS selector:
+
+```
+(CI_TYPE = '{index}' AND method = 'build' AND scratch = FALSE) OR (CI_TYPE = 'ci-metricsdata')
```

Comment:
For convenience sake, {index} was brew-taskstatechange in the listener.  I have since changed that as I was essentially just overusing it.  In the latest version of the listener I have a PR out for, this has changed to CI_TYPE = 'brew-taskstatechange' ...